xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -131,7 +135,11 @@ protected boolean delete(XWikiContext context) throws XWikiException
         XWikiDocument doc = context.getDoc();
 
         String sindex = request.getParameter(RECYCLED_DOCUMENT_ID_PARAM);
-        if (sindex != null && xwiki.hasRecycleBin(context)) {
+        String emptyBin = request.getParameter(EMPTY_RECYCLE_BIN);
+
+        if ("1".equals(emptyBin)) {
```

Comment:
Is there a more generic way to do that? `emptybin=true` should also work right?

Reply:
Yeah it's actually what's used in https://github.com/xwiki/xwiki-platform/pull/928/files/d6dee0437738b231f426a43d00913860d00c9e7f#diff-70855079f42aabba98e87755fa9e71ecR65 I can change that.